### DISCUSSION TITLE
Git push failed

### DISCUSSION THREAD

**Discussion by tsheringtashi** [community/community#23198] [New to GitHub]
<p>I executed git push origin master and I got the following error. failed to push some refs to github.</p>
<p><img src="https://user-images.githubusercontent.com/34204854/181075555-06ce5f50-3550-4263-9b67-cb76a51cfaa7.jpeg" alt="image" data-base62-sha1="cu1lb6DWJedYEAhBmq4gy7Wl33K" width="601" height="335"></p>

**Comment by airtower-luna** [comment]
<p>The error message describes exactly what’s going on: There are commits in your Github repository that are not in your local repository. You’ll need to either merge the remote changes with your local work (if you want to keep both local and remote changes) or override the warning (if you are absolutely sure you want to get rid of the remote-only changes).</p>

**Comment by tsheringtashi** [comment]
<p>thank you for your help. I don’t know how to merge the remote changes with local repository.</p>

**Comment by kingthorin** [comment]
<pre><code class="lang-auto">git fetch upstream
git merge upstream/master
</code></pre>

**Comment by Wabri** [comment, marked as answer]
<p>Or simply use</p>
<pre><code class="lang-auto">git pull
</code></pre>

**Comment by tsheringtashi** [comment]
<p>thank you for your support. I used</p>
<p>git pull origin master.</p>
<p>given reason in editor and saved.</p>
<p>worked!</p>
